client/components/ProductOverview/ImageGallery.jsx

```diff
@@ -1,26 +1,39 @@
-import React, { useState } from 'react';
+import React from 'react';
 import { AiOutlineArrowLeft, AiOutlineArrowRight } from 'react-icons/ai';
 import PropTypes from 'prop-types';
 
-const ImageGallery = ({ currentStyle }) => {
-  const [currentImageIndex, setCurrentImageIndex] = useState(0);
-  const currentPhoto = currentStyle.photos[currentImageIndex].url;
+import ThumbnailView from './ThumbnailView';
+
+const ImageGallery = ({ currentStyle, imageIndex, updateImageIndex }) => {
+  if (currentStyle === null) {
+    return <div>No data available</div>;
+  }
+
   const { photos } = currentStyle;
   const { length } = photos;
+  if (imageIndex > length) {
+    updateImageIndex(0);
+  }
+  const currentPhoto = currentStyle.photos[imageIndex].url;
 
   const handleLeftArrow = () => {
-    setCurrentImageIndex(currentImageIndex === 0 ? length - 1 : currentImageIndex - 1);
+    updateImageIndex(imageIndex === 0 ? length - 1 : imageIndex - 1);
   };
 
   const handleRightArrow = () => {
-    setCurrentImageIndex(currentImageIndex === length - 1 ? 0 : currentImageIndex + 1);
+    updateImageIndex(imageIndex === length - 1 ? 0 : imageIndex + 1);
+  };
+
+  const handleThumbNailClick = (index) => {
+    updateImageIndex(index);
   };
 
   return (
-    <div className="mainImageContainer">
-      <AiOutlineArrowLeft className="leftArrow" onClick={handleLeftArrow} />
-      <AiOutlineArrowRight className="rightArrow" onClick={handleRightArrow} />
-      <img className="mainImage" src={currentPhoto} alt="selected style" />
+    <div className="imageGalleryContainer">
+      <div className="mainImageContainer" style={{ backgroundImage: `url(${currentPhoto})` }} />
+      <ThumbnailView photos={photos} handleClick={handleThumbNailClick} className="thumbnailComponent" currentIndex={imageIndex} />
+      <AiOutlineArrowLeft className="leftArrow" onClick={handleLeftArrow} size={35} />
```

Comment:
Can we create a variable for the size and store it at the top of the component? Something like `const ARROW_SIZE = 35;` That way if anything changes, you'd only have to change it in one place.

Reply:
refactored